File: 프로그래머스/lv3/42628. 이중우선순위큐/이중우선순위큐.java
import java.util.*;

class Solution {
    public int[] solution(String[] operations) {
        PriorityQueue<Integer> minpq=new PriorityQueue<>();
        PriorityQueue<Integer> maxpq=new PriorityQueue<>(Collections.reverseOrder());
        for(int i=0;i<operations.length;i++){
            String[] s=operations[i].split(" ");
            if(s[0].equals("I")){
                minpq.add(Integer.parseInt(s[1]));
                maxpq.add(Integer.parseInt(s[1]));
            }
            else{
                if(minpq.size()==0||maxpq.size()==0){continue;}
                if(s[1].equals("1")){
                    minpq.remove(maxpq.poll());
                }
                else{
                    maxpq.remove(minpq.poll());
                }
            }
            
            
            
        }
        int[] answer=new int[2];
        if(minpq.size()==0){
            answer[0]=0;
            answer[1]=0;
        }
        else{
            answer[0]=maxpq.poll();
            answer[1]=minpq.poll();
        }
        
        return answer;
    }
}

File: 프로그래머스/lv3/42628. 이중우선순위큐/이중우선순위큐.py
def solution(operations):
    answer = []
    for i in operations:
        if i[0]=='I':
            answer.append(int(i[2:]))
        elif len(answer)>0:
            if 'D 1' in i:
                answer.remove(max(answer))
            elif 'D -1' in i:
                answer.remove(min(answer))
    if len(answer)==0:
        return [0,0]
    mi=min(answer); ma=max(answer)
    return [ma,mi]